inst/@sym/private/elementwise_op.m

```diff
@@ -94,15 +94,13 @@
           '    return _op(*_ins)'
           % at least one input was a matrix:
           '# dbout(f"at least one matrix param, shape={q.shape}")'
-          'elements = []'
-	  'assert len(q.shape) == 2, "non-2D arrays/tensors not yet supported"'
-          'for i in range(0, q.shape[0]):'
-          '    for j in range(0, q.shape[1]):'
-          '        elements.append(_op(*[k[i, j] if isinstance(k, (MatrixBase, NDimArray)) else k for k in _ins]))'
-          'if all(isinstance(x, Expr) for x in elements):'
-          '    return Matrix(*q.shape, elements)'
-          'dbout(f"elementwise_op: returning an Array not a Matrix")'
-          'return Array(elements, shape=q.shape)' ];
```

Comment:
note in the old code here we have `elements` a flat list.  Then we'd simply make `make_matrix_or_array(<iterable>, <shape>)`...